js/main.js

```diff
@@ -0,0 +1,41 @@
+var $entryForm = document.forms[0];
+var $image = document.querySelector('.image');
+var $photoUrl = document.querySelector('.photoUrl');
+
+function handleImage(event) {
+  var photoUrlValue = $entryForm.elements.photoUrl.value;
+  $image.setAttribute('src', photoUrlValue);
+}
+
+$photoUrl.addEventListener('input', handleImage);
+
+function handleEntryForm(event) {
+  event.preventDefault();
+  var formValues = {
+    title: $entryForm.elements.title.value,
+    photoUrl: $entryForm.elements.photoUrl.value,
+    notes: $entryForm.elements.notes.value
+  };
+  formValues.entryId = data.nextEntryId;
+  data.nextEntryId++;
+  data.entries.unshift(formValues);
+  $image.setAttribute('src', '/images/placeholder-image-square.jpg');
+  $entryForm.elements.title.value = '';
+  $entryForm.elements.photoUrl.value = '';
+  $entryForm.elements.notes.value = '';
+}
+
+$entryForm.addEventListener('submit', handleEntryForm);
+
+var previousDataJSON = localStorage.getItem('javascript-local-storage');
+
+if (previousDataJSON !== null) {
+  JSON.parse(previousDataJSON);
```

Comment:
You should be assigning this to `data`

Reply:
https://user-images.githubusercontent.com/101234537/179651474-857245e4-465c-4502-a8e5-ab17fb67b6d6.mov